### PR TITLE
Prepare 0.101.0 release

### LINKED ISSUE #3162: Prepare 0.101.0 release
Check list:

PR: #3163

- [x] version updated in `pyproject.toml`
- [x] `DEV=False` in `__init__.py`
- [x] pip installs from `git` commented out
- [x] full tests passing
- [x] documentation builds
- [x] all documentation tutorials and how tos without errors
- [ ] new functions in API reference
- [x] new features in docs
- [x] README points to new stable documentation
- [x] release notes in `doc/releases`
- [x] summary of release in `doc/whatisnew.rst`
- [ ] All milestone PRs merged

### PR DESCRIPTION
**Pull request by alejoe91**
Closes #3162

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,11 +46,13 @@
 [![Twitter](https://img.shields.io/badge/@spikeinterface-%231DA1F2.svg?style=for-the-badge&logo=Twitter&logoColor=white)](https://twitter.com/spikeinterface) [![Mastodon](https://img.shields.io/badge/-@spikeinterface-%232B90D9?style=for-the-badge&logo=mastodon&logoColor=white)](https://fosstodon.org/@spikeinterface)
 
 
-> :warning::warning::warning:
-> **New features under construction!** 🚧🚧🚧: after the 0.100.0 release (and related bug fixes), the next release will contain
-> a major API improvement: the `SortingAnalyzer`. To read more about this, checkout the
-> [enhancement proposal](https://github.com/SpikeInterface/spikeinterface/issues/2282).
-> Please refer to the stable documentation [here](https://spikeinterface.readthedocs.io/en/0.100.4)
+> :rocket::rocket::rocket:
```
Comment (zm711) [inline README.md:49]: :)

**Review by zm711 — COMMENTED** [review]
One tiny change I think we can make which reduces the burdening in updating the docs now :)

File: doc/releases/0.101.0.rst [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+13th July 2024
+
+Main changes:
+
+* Implementation of `SortingAnalyzer` (#2398)
+* Added framework for hybrid recording generation (#2436, #2769, #2857)
+* Improved auto-merging functions and enable `SortingAnalyzer` to merge units and extensions (#2934, #3043, #3154)
+* Refactored motion correction with addition of DREDGE AP and LFP (#2915, #3062)
+* Better handling of time information (#3118, #3119, #3120)
```
Comment (samuelgarcia) [inline doc/releases/0.101.0.rst:14]: + benchmark components

File: doc/releases/0.101.0.rst [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+------------------------------------
+
+13th July 2024
+
+Main changes:
+
+* Implementation of `SortingAnalyzer` (#2398)
+* Added framework for hybrid recording generation (#2436, #2769, #2857)
+* Improved auto-merging functions and enable `SortingAnalyzer` to merge units and extensions (#2934, #3043, #3154)
+* Refactored motion correction with addition of DREDGE AP and LFP (#2915, #3062)
```
Comment (samuelgarcia) [inline doc/releases/0.101.0.rst:13]: +Motion object

File: doc/releases/0.101.0.rst (hunk 2 above)
Comment (samuelgarcia) [reply doc/releases/0.101.0.rst:14]: * generate drift also

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**